astropy/io/fits/card.py

```diff
@@ -1299,20 +1299,21 @@ def _format_value(value):
 
 def _format_float(value):
     """Format a floating number to make sure it gets the decimal point."""
-    value_str = f"{value:.16G}"
-    if "." not in value_str and "E" not in value_str:
-        value_str += ".0"
-    elif "E" in value_str:
+    value_str = str(value)
```

Comment:
I think we may have to do `str(value).upper()` to ensure we still have an `E` - that may well be FITS standard.

Reply:
I've changed the if and .split() to check for "e" instead of "E" since that is how python represents the exponent. I believe the upper case E is indeed part of the FITS standard, however that is not an issue since I did not change the following line:

https://github.com/astropy/astropy/blob/60835aa8b73f64a7a4402a000605d2d0fa1105b9/astropy/io/fits/card.py#L1314

This will force the end result to always use the upper case E.

Reply:
Good point. So, then it is more relevant if we actually can remove that whole stanza (which I think is likely to be the case).

Reply:
Sounds a bit risky but we can give it a try. I notice that they specifically have `int(exponent)`, but I don't think it's even possible that Python ever returns a fractional exponent (`1e0.5` isn't even legal syntax). Also, `1e-9` is converted to `"1e-09" ` so it seems it automatically pads to at least 2 digits. The only way this would fail is if indeed the representation is different in different platforms.

Reply:
This part looks very sketchy to me:
https://github.com/astropy/astropy/blob/60835aa8b73f64a7a4402a000605d2d0fa1105b9/astropy/io/fits/card.py#L1315-L1316

The only way this `if` statement would be called is if a integer reached this point, but I don't think that could ever happen because we only call this function in these places:
https://github.com/astropy/astropy/blob/60835aa8b73f64a7a4402a000605d2d0fa1105b9/astropy/io/fits/card.py#L1287-L1292

I've checked and using both Python built-in complex numbers and numpy's complex numbers, `.real` and `.imag` are float even if we use `1j+1`. In other words, I don't see how we could get a number that does not have a decimal point while also not using an exponent.

Reply:
Agreed, that can be removed too! It was needed only with the format string:
```
In [2]: f"{5.0:16G}"
Out[2]: '               5'
```